__tests__/_helper.ts

```diff
@@ -0,0 +1,140 @@
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2020 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/serlo.org-cloudflare-worker for the canonical source repository
+ */
+export async function containsText(response: Response, texts: string[]) {
+  expect(response).not.toBeNull()
+
+  const responseText = await response.text()
+  texts.forEach((text) =>
+    expect(responseText).toEqual(expect.stringContaining(text))
+  )
+}
+
+export function contentTypeIsHtml(response: Response): void {
+  expect(response.headers.get('Content-Type')).toBe('text/html;charset=utf-8')
+}
+
+export function hasOkStatus(response: Response): void {
+  expect(response).not.toBeNull()
+  expect(response.status).toBe(200)
+  expect(response.statusText).toBe('OK')
+}
+
+export async function isNotFoundResponse(response: Response): Promise<void> {
+  expect(response).not.toBeNull()
+  expect(response.status).toBe(404)
+  expect(response.statusText).toBe('Not Found')
+  expect(await response.text()).toEqual(
+    expect.stringContaining('Page not found')
+  )
+}
+
+export async function isJsonResponse(response: Response, targetJson: unknown) {
+  hasOkStatus(response)
+  expect(response.headers.get('Content-Type')).toBe('application/json')
+  expect(JSON.parse(await response.text())).toEqual(targetJson)
+}
+
+type FetchSpec = Record<string, string | Response>
+
+export function mockFetch(spec: FetchSpec = {}): FetchMock {
+  const mockedFetch = jest.fn()
+  const mockedInternet = new FetchMock(spec, mockedFetch)
+
+  mockedFetch.mockImplementation((reqInfo) => mockedInternet.fetch(reqInfo))
+
+  global.fetch = mockedFetch
+
+  return mockedInternet
+}
+
+export class FetchMock {
+  private specs: FetchSpec
+  private mockedFetch: jest.Mock
+
+  constructor(specs: FetchSpec, mockedFetch: jest.Mock) {
+    this.specs = specs
+    this.mockedFetch = mockedFetch
+  }
+
+  get mock() {
+    return this.mockedFetch
+  }
+
+  fetch(reqInfo: Request | string) {
+    const url = typeof reqInfo === 'string' ? reqInfo : reqInfo.url
+    const responseSpec = this.specs[url]
+    const errorMessage = `Response for URL ${url} is not defined in mocked fetch`
+
+    return responseSpec === undefined
+      ? Promise.reject(new Error(errorMessage))
+      : Promise.resolve(FetchMock.convertToResponse(responseSpec))
+  }
+
+  addResponseFor(url: string, spec: string | Response = '') {
```

Comment:
addResponseFor('url')` without second argument is kinda weird. Maybe
```suggestion
  mockRequests({ to, response = '' }: { to: string, response: string | Response }) {
```
reads nicer:
```ts
fetchApi.mockRequest({ to: 'http://api.serlo.org/graphql' }) // no specific response
fetchApi.mockRequest({ to: 'http://api.serlo.org/graphql', response: 'foobar' }) // specific response
```